CMakeLists.txt

```diff
@@ -157,14 +157,27 @@ add_subdirectory(test)
 if(ENABLE_PYTHON_BINDINGS)
   find_package(PythonInterp)
   find_package(PythonLibs)
-  if(${PYTHON_VERSION_MAJOR} EQUAL 3)
+
+  ## This a workaround for CMake Boost::python configuration
+  if(NOT ${PYTHON_VERSION_MAJOR} EQUAL 3)
+    set(Boost_Python_lib "Boost::python")
+    find_package(Boost COMPONENTS python)
+  endif()
+
+  if(NOT Boost_PYTHON_FOUND)
     set(Boost_Python_lib "Boost::python${PYTHON_VERSION_MAJOR}")
     set(_Boost_PYTHON${PYTHON_VERSION_MAJOR}_HEADERS "boost/python.hpp")
     find_package(Boost COMPONENTS python${PYTHON_VERSION_MAJOR})
-  else()
-    set(Boost_Python_lib "Boost::python")
-    find_package(Boost COMPONENTS python)
+    if(NOT Boost_PYTHON${PYTHON_VERSION_MAJOR}_FOUND)
+      set(Boost_Python_lib "Boost::python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+      set(_Boost_PYTHON${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}_HEADERS "boost/python.hpp")
+      find_package(Boost COMPONENTS python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+      if(NOT Boost_PYTHON${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}_FOUND)
+        message(SEND_ERROR "Boost::python is not found")
+      endif()
+    endif()
```

Comment:
This is more comprehensive and less brittle than what I had done to get it working locally. 👍